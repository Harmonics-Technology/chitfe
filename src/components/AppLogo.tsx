import Image from 'next/image';
import Link from 'next/link';

import ChitLogo from '~/public/chit.svg';

export default function AppLogo() {
    return (
        <Link href='/' as='image'>
            <Image
                priority
                src={ChitLogo}
                alt='Chit brand logo'
                className='w-20 lg:w-36'
            />
        </Link>
    );
}
