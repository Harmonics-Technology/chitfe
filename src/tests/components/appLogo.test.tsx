import { screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import { render } from '../test-utils';
import AppLogo from '@components/app-logo';

jest.mock('next/link', () => {
    return ({
        children,
        href,
    }: {
        children: React.ReactNode;
        href: string;
    }) => <a href={href}>{children}</a>;
});

jest.mock(
    'next/image',
    () => (props: React.ImgHTMLAttributes<HTMLImageElement>) => {
        // const { ...rest } = props;
        return <img {...props} />;
    }
);

describe('AppLogo', () => {
    it('renders the logo', () => {
        const { container } = render(<AppLogo />);
        const logo = screen.getByAltText('Chit brand logo');

        expect(logo).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });

    it('renders the icon when isIcon is true', () => {
        render(<AppLogo isIcon={true} />);

        const icon = screen.getByAltText('Chit brand icon');
        expect(icon).toBeInTheDocument();
    });

    it("wraps the image in a link pointing to '/'", async () => {
        render(<AppLogo />);

        const link = screen.getByRole('link');
        expect(link).toHaveAttribute('href', '/');
    });
});
