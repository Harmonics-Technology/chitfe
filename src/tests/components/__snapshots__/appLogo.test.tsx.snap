// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLogo renders the logo 1`] = `
<div>
  <a
    href="/"
  >
    <img
      alt="Chit brand logo"
      class="w-28 lg:w-36"
      src="[object Object]"
    />
  </a>
</div>
`;
